cmd/template/DBDriver/postgres.go

```diff
@@ -0,0 +1,41 @@
+package DBDriver
+
+type PostgresTemplate struct{}
+
+func (m PostgresTemplate) Service() []byte {
+	return []byte(`package services
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	_ "github.com/lib/pq"
+	"log"
+	"time"
+)
+
+type Service struct {
+	db *sql.DB
+}
+
+func New() *Service {
```

Comment:
Recommend using interfaces for mockable services
```go
type Service interface {
	Health()
}

type service struct {
	db *sql.DB
}

func New() *service {
	// ...
}
```

Reply:
Wouldn't the IDE start yelling that we are trying to return an unexported type here?

Reply:
Nope. The calling code would depend on the interface rather than the concrete struct type.

```go
type healthHandler struct {
	s services.Service
}

func NewHealthHandler() *healthHandler {
	return &handler{
		s: services.New(),
	}
}

func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request)) {
	w.Write(h.s.Health())
}
```

Reply:
I would appreciate it if you could take a look now. I wonder if this could be implemented in a more suitable way?